src/main/java/com/google/devtools/build/lib/bazel/repository/skylark/SkylarkRepositoryContext.java

```diff
@@ -262,6 +263,29 @@ public void createFileFromTemplate(
     }
   }
 
+  @Override
+  public String readFile(Object path, String encoding, Location location)
+      throws RepositoryFunctionException, EvalException, InterruptedException {
+    SkylarkPath p = getPath("read()", path);
+    Charset charset;
+    if (encoding.equals("")) {
+      charset = StandardCharsets.ISO_8859_1;
```

Comment:
We want 8859-1 as the default?

Reply:
The default is set to UTF-8 in `SkylarkRepositoryContextApi`. I'm using `StandardCharsets.ISO_8859_1` to mean raw bytes, for when the user calls `ctx.read("somefile", encoding="")`.

My assumption is that this function will almost always be used to ingest text files in ASCII / UTF-8, with `encoding = ""` available as an escape hatch for odd cases.

Reply:
Oh, I see!  Ok.  Gotcha.